guides/elixir.md

```diff
@@ -0,0 +1,937 @@
+[← Avvo Coding Styleguides](https://github.com/avvo/coding-style)
+
+---
+
+
+This could be considered as a minimal set of conventions [approved by José Valim](https://elixirforum.com/t/the-elixir-style-guide/2534/7) 
+
+(forked from [lexmax's style guide](https://github.com/lexmag/elixir-style-guide))
+
+
+_You might also want to check out more broader and detailed community driven style guide for Elixir by [Christopher Adams](https://github.com/christopheradams/elixir_style_guide)_
+
+# Elixir Style Guide
+
+> A programmer does not primarily write code; rather, he primarily writes to another programmer about his problem solution. The understanding of this fact is the final step in his maturation as technician.
+>
+> — <cite>What a Programmer Does, 1967</cite>
+
+## Table of Contents
+
+* [Linting](#linting)
+  * [Naming](#naming)
+  * [Comments](#comments)
+  * [Modules](#modules)
+  * [Regular Expressions](#regular-expressions)
+  * [Structs](#structs)
+  * [Exceptions](#exceptions)
+  * [ExUnit](#exunit)
+
+The following section are automatically applied by the code formatter in Elixir v1.6 and listed here only for documentation purposes:
+
+* [Formatting](#formatting)
+  * [Whitespace](#whitespace)
+  * [Indentation](#indentation)
+  * [Term representation](#term-representation)
+  * [Parentheses](#parentheses)
+  * [Layout](#layout)
+
+## Linting
+
+* <a name="pipeline-operator"></a>
+  Favor the pipeline operator `|>` to chain function calls together.
+  <sup>[[link](#pipeline-operator)]</sup>
+
+  ```elixir
+  # Bad
+  String.downcase(String.strip(input))
+
+  # Good
+  input |> String.strip() |> String.downcase()
+  ```
+
+  For a multi-line pipeline, place each function call on a new line, and retain the level of indentation.
+
+  ```elixir
+  input
+  |> String.strip()
+  |> String.downcase()
+  |> String.slice(1, 3)
```

Comment:
This example is a little confusing because it would all fit on one line. It took me a minute to figure out what the difference between the "good" example in the chaining section above was and this one.

Reply:
Oh, I think you misunderstood this part.
It says if you have a chain of function calls (more than one) - just prefer `|>`.. ether one line or multi-line.. but if you choose to go with multiline it just says that make sure that every function call is on a new line and level of indentation is the same.
For example don't do this:
```elixir 
# bad
input
  |> String.strip()
  |> String.downcase()
  |> String.slice(1, 3)
```

Anyway since elixir 1.6 `mix format` will handle indentation